proposals/NNNN-SDLScreen-SDLWindow-Projection.md

```diff
@@ -1 +1,49 @@

```objc
extern NSString *const SDLVideoStreamDidStartNotification;
extern NSString *const SDLVideoStreamDidStopNotification;

extern NSString *const SDLAudioStreamDidStartNotification;
extern NSString *const SDLAudioStreamDidStopNotification;

@class SDLManager;

@interface SDLStreamingMediaLifecycleManager : NSObject

@property (nonatomic, weak, nullable) SDLManager *sdlManager;

@property (nonatomic, assign, readonly, getter=isVideoConnected) BOOL videoConnected;
@property (nonatomic, assign, readonly, getter=isAudioConnected) BOOL audioConnected;

@property (nonatomic, assign, readonly, getter=isVideoStreamingAllowed) BOOL videoStreamingAllowed;

@end
```

## Potential downsides

In subclassing UIScreen and UIWindow, there may be undocumented and unforseen interactions between iOS and the subclassed objects. As new iOS versions come out, bugs might emerge due to changes Apple makes behind the scenes of these public interfaces.

## Impact on existing code

This proposal has no impact on existing video/audio streaming applications. It provides a new higher level interface that should improve developer productivity and a foundation for both touch and physical button focus/select interactions.

## Alternatives considered

The SDLScreen and SDLWindow classes could be derived from NSObject instead. This may be preferable.


\ No newline at end of file
+# SDLCarWindow Video Projection Developer Interface

* Proposal: [SDL-NNNN](NNNN-SDLScreen-SDLWindow-Projection.md)
* Author: [David Switzer](https://github.com/davidswi)
* Status: **Awaiting review**
* Impacted Platforms: [iOS]

## Introduction

The iOS SDL proxy exposes the SDLCarWindow class to provide an easy-to-integrate developer interface to SDL video projection and the associated task of defining focusable and selectable UI elements. The SDLCarWindow attaches to the SDLStreamingMediaManager and exposes the SDLInterfaceManager described in the [SDLInterfaceManager](https://github.com/Toyota-BSalahat/sdl_evolution/tree/SDLInterfaceManager/proposals) proposal to allow the latter to determine the spatial arrangement of controls. The SDLInterfaceManager adopts the [SDLHapticHitTester] (https://github.com/smartdevicelink/sdl_evolution/issues/270) protocol to enable correlation of SDLTouch events with the corresponding haptic region. In a later proposal, the SDLInterfaceManager will be extended to include focus and selection control on the handset, following the UIFocusEngine model from tvOS.

## Motivation

This proposal builds on the SDLStreamingMediaManager to provide a convenient and simple programming model for SDL app developers to project and enable interaction with the appropriate subset of their UIKit, OpenGL and hybrid UIKit/OpenGL user interfaces.

## Proposed solution

The SDLStreamingMediaManager class is extended to include a car window property:

```objc  
@interface SDLStreamingMediaManager : NSObject <SDLProtocolListener>
```

Comment:
The SDLInterfaceManager link seems to be broken.  This should link here: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0081-SDLInterfaceManager.md.

"SDLHapticHitTester" is not linked correctly.  You may need to remove a space between the ] and (